jetty-server/src/main/java/org/eclipse/jetty/server/AbstractNCSARequestLog.java

```diff
@@ -489,4 +496,58 @@ public String getLogTimeZone()
     {
         return _logTimeZone;
     }
+    
+
+    private String extractUser(Request request)
```

Comment:
I think this change is too special purpose to put into a general jetty release.  My concerns are that the extracted auth header has not actually been authenticated nor authorised.  

So can you change the PR, so that the extractUser method is protected and just does handles the existing normal case.  This then gives you a very simple method that you can extend in your own specialisation of the log and you can use.